astropy/io/fits/_utils.pyx

```diff
@@ -0,0 +1,57 @@
+from collections import OrderedDict
+
+cdef Py_ssize_t BLOCK_SIZE = 2880  # the FITS block size
+cdef Py_ssize_t CARD_LENGTH = 80
+cdef str VALUE_INDICATOR = '= '  # The standard FITS value indicator
+cdef str END_CARD = 'END' + ' ' * 77
+
+
+def parse_header(fileobj):
+    """The main method allowing to parse quickly a FITS header."""
+
+    cards = OrderedDict()
+    read_blocks = []
+    cdef int found_end = 0
+    cdef bytes block
+    cdef str header_str, block_str, card_image, keyword
+    cdef Py_ssize_t idx, end_idx, sep_idx
+
+    while True:
+        # iterate on blocks
+        block = fileobj.read(BLOCK_SIZE)
+        if not block or len(block) < BLOCK_SIZE:
+            # header looks incorrect, raising exception to fall back to
+            # the full Header parsing
+            raise Exception
+
+        block_str = block.decode('ascii')
+        read_blocks.append(block_str)
+        idx = 0
+        while idx < BLOCK_SIZE:
+            # iterate on cards
+            end_idx = idx + CARD_LENGTH
+            card_image = block_str[idx:end_idx]
+            idx = end_idx
+
+            keyword = card_image[:8].strip()
+            sep_idx = card_image.find(VALUE_INDICATOR)
```

Comment:
Ins't the ``=`` separator always supposed to be in a specific character position? ``find`` could be slow... Ah I see what comes below. What about, to speed things up in the most common case, making the start of the 'if' statement ``if card_image[8] == '='`` and only using ``find`` if that fails?

Reply:
Good idea, I will try this.

Reply:
[`find`](https://docs.python.org/3/library/stdtypes.html#str.find) supports start and stop, maybe we can use these to limit the "searched region": `card_image.find(VALUE_INDICATOR, 0, 9)` (not sure if the indices are correct ... 😓)

Reply:
Good to know. I already implemented something with @astrofrog 's suggestion (and forgot to push it ...), where I slice the first 8 characters before the `.find`. Not sure if it makes a big difference with using the start and stop parameters.

Reply:
>Not sure if it makes a big difference with using the start and stop parameters.

It's probably faster because it avoids the copy. Probably not big but worth a try

Reply:
It seems to make a small difference, always good to take :+1: